Library/ViewModels/ProjectPamphletContentViewModelTests.swift

```diff
@@ -698,5 +702,225 @@ final class ProjectPamphletContentViewModelTests: TestCase {
     }
   }
 
+  // swiftlint:disable line_length
+  func testLoadProjectIntoDataSource_ProjectSummaryLoaded_ExperimentalVariant() {
+    let project = Project.template
+
+    let envelope = ProjectSummaryEnvelope(projectSummary: [
+      .init(question: .whatIsTheProject, response: "response-1"),
+      .init(question: .whatWillYouDoWithTheMoney, response: "response-2"),
+      .init(question: .whoAreYou, response: "response-3")
+    ])
+
+    let mockService = MockService(fetchProjectSummaryResult: .success(envelope))
+
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeMeProjectSummary.rawValue: OptimizelyExperiment.Variant.variant1.rawValue
+      ]
+
+    withEnvironment(apiService: mockService, optimizelyClient: optimizelyClient) {
+      self.vm.inputs.configureWith(value: (project, .discovery))
+      self.vm.inputs.viewDidLoad()
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceProject.assertValues([], "Nothing emits immediately.")
+      self.loadProjectIntoDataSourceRefTag.assertValues([], "Nothing emits immediately.")
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing emits immediately.")
+
+      // Begin presentation. When presenting the project `animated` will be false since it is embedded in the
+      // navigator controller.
+      self.vm.inputs.viewWillAppear(animated: false)
+      self.vm.inputs.viewDidAppear(animated: false)
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([[]], "Emits [] as view appears")
+      self.loadProjectIntoDataSourceProject
+        .assertValues([project], "Load the full project into the data source.")
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery], "Load the refTag into the data source."
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues(
+        [], "Do not load the minimal version of the project."
+      )
+
+      // End presentation.
+      self.vm.inputs.viewWillAppear(animated: false)
+      self.vm.inputs.viewDidAppear(animated: false)
+
+      self.loadProjectIntoDataSourceProjectSummaryItems.assertValues([[]])
+      self.loadProjectIntoDataSourceProject
+        .assertValues([project], "Nothing new emits when the view is done.")
+      self.loadProjectIntoDataSourceRefTag.assertValues(
+        [.discovery], "Nothing new emits when the view is done."
+      )
+      self.loadMinimalProjectIntoDataSource.assertValues([], "Nothing new emits when the view is done.")
+
+      // Simulate a new version of the project coming through
+      self.vm.inputs.configureWith(value: (project, .discovery))
```

Comment:
Does this ever happen IRL?

Reply:
Yeah the project page is first configured with a minimal version and then with the updated project that includes more data.